client/login/wp-login/index.jsx

```diff
@@ -104,12 +118,16 @@ class Login extends React.Component {
 				</a>;
 		}
 
-		const goBackLink = ! magicLoginView && <a
-			href="#"
-			key="back-link"
-			onClick={ this.goBack }>
-				<Gridicon icon="arrow-left" size={ 18 } /> { this.props.translate( 'Return' ) }
-			</a>;
+		let goBackLink;
+		if ( this.state.loaded && window.history.length > 1 ) {
```

Comment:
It gives that warning because on SSR there's no `window`, but when you render on client there is a history, hence the warning of mismatched markup.